### DISCUSSION TITLE
TailwindCss group hovers does not work

### DISCUSSION THREAD

**Discussion by irorochad** [tailwindlabs/tailwindcss#10425] [Help]
Hi, @all  I'm trying to add group hover to a button but it seems I'm not getting it well.

`
    
        <div>
          <button
            type="button"
            className="group px-12 py-5 hover:scale-btn bg-rb-light-200 dark:bg-rb-dark-0 text-active dark:text-rb-bg-200 font- 
                bold text-2xl leading-normal rounded-xl hover:scale-x-125 hover:bg-rb-btn-hover hover:text-rb-light-200 
                hover:shadow-lg  transition duration-150 ease-in-out flex justify-between"
          >
            <span > GET STARTED - IT&apos;S FREE</span>
            <span className="invisible group-hover:visible">
              < An invisible icon is here, but I want it shown when the group hovers. />
            </span>
          </button>
       </div>
        
`

If you copy this code to play.tailwindcss.com, it'd work.  

I know I have to set the variant but I don't really know how to use it well, I guess.

Take a look at my variant in config file

` 

   variants: {
      extend: {
        scale: ['hover', 'active', 'group-hover'],
        button: ['hover', 'group-hover'],
      },

`

Please let me know if you can help me.

Thanks.

**Comment by wongjn** [comment]
I am assuming that this is in regards to the inner `<span>` element with the `group-hover:visible` class. I'll also assume you're using Tailwind v2 since you mention `variant` configuration.

The HTML would work in Tailwind Play if you forgot to change the version of Tailwind in it and left it at v3.

It seems like you're missing the `group-hover` variant for `visibility`:
```diff
 variants: {
   extend: {
     scale: ['hover', 'active', 'group-hover'],
     button: ['hover', 'group-hover'],
+    visibility: ['group-hover'],
   },
 },
```
[In Tailwind Play](https://play.tailwindcss.com/RbJTnlGhc0)

**Reply by irorochad** [reply]
@wongjn Thanks for the reply. I just found out that I was using the version 2 of tailwind. I have upgraded to version 3. All working fine.

Thanks!
